### ISSUE TITLE
[HOWTO] Keep treemacs open when pressing `esc`

### ISSUE THREAD

**Issue by tam5** [hlissner/doom-emacs#1970]
**What I want to achieve**
I'd like to keep treemacs open when I press escape (`evil-force-normal-state`), so that treemacs is only toggled when I explicitly toggle it via `+treemacs/toggle`.

Is there an easy/good way to do this?

**System information**
<details><pre>
<details>

```
emacs   version    26.2
        features   NOTIFY ACL GNUTLS LIBXML2 ZLIB TOOLKIT_SCROLL_BARS NS MODULES THREADS
        build      Apr 12, 2019
        buildopts  (--with-ns '--enable-locallisppath=/Library/Application Support/Emacs/${version}/site-lisp:/Library/Application Support/Emacs/site-lisp' --with-modules)
        windowsys  ns
        daemonp    server-running
doom    version    2.0.9
        build      HEAD -> develop, origin/develop, origin/HEAD 23ef6e42c 2019-10-27 11:40:47 -0400
system  type       darwin
        config     x86_64-apple-darwin18.2.0
        shell      /bin/bash
        uname      Darwin 19.0.0 Darwin Kernel Version 19.0.0: Wed Sep 25 20:18:50 PDT 2019; root:xnu-6153.11.26~2/RELEASE_X86_64 x86_64
        path       (/usr/local/opt/ruby/bin/ ~/.composer/vendor/bin/ ~/Code/platform/ ~/.emacs.d/bin/ ~/bin/ /usr/local/bin/ /usr/bin/ /bin/ /usr/sbin/ /sbin/ /Applications/Emacs.app/Contents/MacOS/bin-x86_64-10_14/ /Applications/Emacs.app/Contents/MacOS/libexec-x86_64-10_14/ /Applications/Emacs.app/Contents/MacOS/libexec/)
config  envfile    envvar-file
        elc-files  0
        modules    (:completion (company +auto +childframe) (ivy +fuzzy) tab-complete :ui doom doom-dashboard doom-quit ophints workspaces hl-todo indent-guides modeline nav-flash treemacs (popup +all +defaults) vc-gutter vi-tilde-fringe window-select better-company :editor (evil +everywhere) fold file-templates snippets multiple-cursors rotate-text :emacs dired electric vc :tools (eval +overlay) docker (lookup +docsets) editorconfig flycheck flyspell lsp magit :lang data emacs-lisp go (java +lsp) (javascript +lsp) markdown (org +attach +babel +capture +export +present +protocol) (php +lsp) python rest ruby (scala +lsp) (sh +fish) web :config (default +bindings +smartparens))
        packages   (n/a)
        elpa       (n/a)
```
</details>
</pre></details>

**Comment by hlissner** [comment]
Unless you created a new popup rules for treemacs without a `:quit nil` property, this is already [the default behavior](https://github.com/hlissner/doom-emacs/blob/develop/modules/ui/treemacs/config.el#L14). I cannot reproduce it otherwise.

**Comment by tam5** [comment]
@hlissner can i first of all say you are amazing for this project? We all literally spend hours and hours every day benefiting from your work here, and you responded within 3 mintues! So yeah thanks..

Anyway, you are right, with a fresh install it seems this is the behavior, so I'll have to see what I have conflicting with it.

**Comment by tam5** [comment]
On second thought, I've been able to reproduce the issue with a vanilla doom installation (with the only addition of treemacs).

Steps to reproduce:
1. treemacs toggle
2. focus back in editor buffer
3. create a split (either vertical or horizontal)
4. press escape a few times

Doing this will close treemacs, while I would expect treemacs to remain open.

I believe the issue to have something to do with the balance windows function, but I don't know for sure

**Comment by hlissner** [comment]
This should be fixed in 4fe24df. Let me know if that isn't the case and I'll reopen it. Thanks for letting me know!

**Comment by tam5** [comment]
Better, thanks!
